spec/lib/miq_automation_engine/service_methods/miq_ae_service_service_spec.rb

```diff
@@ -187,17 +187,56 @@ def invoke_ae
     end
 
     it "#retires_on - today" do
-      service_service.retires_on = Date.today
+      service.update_attributes(:retirement_last_warn => Date.today)
+      service_service.retires_on = Time.zone.today
       service.reload
-
+      expect(service.retirement_last_warn).to be_nil
       expect(service.retirement_due?).to be_truthy
     end
 
     it "#retires_on - tomorrow" do
-      service_service.retires_on = Date.today + 1
+      service.update_attributes(
+        :retired              => true,
+        :retirement_last_warn => Time.zone.today,
+        :retirement_state     => "retiring"
+      )
+      service_service.retires_on = Time.zone.today + 1
+      service.reload
+
+      expect(service).to have_attributes(
+        :retirement_last_warn => nil,
+        :retired              => false,
+        :retirement_state     => nil,
+        :retirement_due?      => false
+      )
+    end
+
+    it "#extend_retires_on - no retirement date set" do
+      extend_days = 7
+      service_service.extend_retires_on(extend_days)
+      service.reload
+
```

Comment:
@tinaafitz @jrafanie 
Could this have sporadic failures when the test runs say at 11:59:59 PM?